org.eclipse.jdt.core.tests.model/src/org/eclipse/jdt/core/tests/model/JavaSearchBugs16Tests.java

```diff
@@ -621,6 +623,41 @@ public void testBug574870_4() throws CoreException {
 
 			}
 		}
+
+	/*
+	 * An inner record in a JRE class causes an AIOOBE during a call hierarchy computation.
```

Comment:
Could we improve the comment here ? Even though description mentions about call hierarchy, we are testing a reference search isn't it ? And also the name `RecordClass` is not clear here since there is no java record involvement.